api/offer_template_test.go

```diff
@@ -237,33 +248,114 @@ var _ = Describe("Offer Template Handler", func() {
 			Expect(recorder.Body.String()).To(Equal(templateID))
 		})
 
-		It("should enable an enabled offer template", func() {
+		It("returns status code of 500 if database is unavailable", func() {
+			templateID := "dd21ec96-2890-4ba0-b8e2-40ea67196990"
+			url := "/templates/" + templateID + "/enable"
+			offerTemplateReader := JSONFor(JSON{})
+			request, _ := http.NewRequest("PUT", url, offerTemplateReader)
+
+			oldDB := app.DB
+			db, err := GetTestDB()
+			Expect(err).NotTo(HaveOccurred())
+			app.DB = db
+			app.DB.(*runner.DB).DB.Close() // make DB connection unavailable
+			app.Router.ServeHTTP(recorder, request)
+
+			Expect(recorder.Code).To(Equal(http.StatusInternalServerError))
+			var obj map[string]interface{}
+			err = json.Unmarshal([]byte(recorder.Body.String()), &obj)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(obj["code"]).To(Equal("OFF-004"))
+			Expect(obj["error"]).To(Equal("Update offer template failed"))
+			Expect(obj["description"]).To(Equal("sql: database is closed"))
+			app.DB = oldDB // avoid errors in after each
+		})
+
+		It("should return status code 404 if id doesn't exist", func() {
+			//Given
+			templateID := uuid.NewV4().String()
+			url := "/templates/" + templateID + "/enable"
+			offerTemplateReader := JSONFor(JSON{})
+			request, _ := http.NewRequest("PUT", url, offerTemplateReader)
+
+			//When
+			app.Router.ServeHTTP(recorder, request)
+
+			//Then
+			Expect(recorder.Code).To(Equal(http.StatusNotFound))
+			var obj map[string]interface{}
+			err := json.Unmarshal([]byte(recorder.Body.String()), &obj)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(obj["code"]).To(Equal("OFF-001"))
+			Expect(obj["error"]).To(Equal("OfferTemplateNotFoundError"))
+			Expect(obj["description"]).To(Equal("OfferTemplate was not found with specified filters."))
+		})
+
+		It("should return status code 422 if invalid parameters", func() {
+			//Given
+			templateID := "not-uuid"
+			url := "/templates/" + templateID + "/enable"
```

Comment:
You are not using templateID anywhere else, just create the url:

`url := "/templates/not-uuid/enable`